Sources/Fluent/Relations/Siblings.swift

```diff
@@ -1,47 +1,83 @@
-public final class Siblings<T: Entity> {
-    fileprivate let _query: Query<T>
-    fileprivate let _left: Entity.Type
+/// Represents a many to many relationship
+/// through a Pivot table from the Local 
+/// entity to the Foreign entity.
+public final class Siblings<
+    Local: Entity, Foreign: Entity
+> {
+    /// The id of the Local entity.
+    ///
+    /// This will be used to filter the 
+    /// collection of foreign entities related
+    /// to the local entity type.
+    let id: Node
 
-    public let localKey: String
-    public let foreignKey: String
+    /// Foreign entity key that is foreign
+    /// to the entity it resides on
+    ///
+    /// This id is a pointer to "Local" keys
+    ///
+    /// ex: "foo_id"
+    let foreignKey: String
 
-    public init<E: Entity>(entity: E, localKey: String?, foreignKey: String?) throws {
+    /// Foreign entity key that is local
+    /// to the entity it resides on
+    ///
+    /// "Foreign" keys point to this id.
+    ///
+    /// ex: "id"
+    let localKey: String
+
+    /// Create a new Siblings relationsip using 
+    /// a Local and Foreign entity.
+    ///
+    /// See Siblings.localKey and Siblings.foreignKey
+    /// for more information about how to use them.
+    public init(
+        _ entity: Local,
+        foreignKey: String = Foreign.foreignIdKey,
+        localKey: String = Foreign.idKey
+    ) throws {
         guard let ident = entity.id else {
             throw RelationError.noIdentifier
         }
 
-        let query = try T.query()
-
-        let localKey = localKey ?? T.idKey 
-        let foreignKey = foreignKey ?? "\(T.name)_\(T.idKey)"
-
-        self.localKey = localKey
+        id = ident
         self.foreignKey = foreignKey
-        
-        self._left = E.self
+        self.localKey = localKey
+    }
+}
 
-        let pivot = Pivot<E, T>.self
+extension Siblings: QueryRepresentable {
+    /// Creates a Query from the Siblings relation.
+    /// This includes a pivot, join, and filter.
+    public func makeQuery() throws -> Query<Foreign> {
+        let query = try Foreign.query()
+
+        let pivot = Pivot<Local, Foreign>.self
 
-        try query.union(
+        try query.join(
             pivot,
-            localKey: localKey,
-            foreignKey: foreignKey
+            localKey: foreignKey,
+            foreignKey: localKey
         )
 
-        try query.filter(pivot, "\(E.name)_\(E.idKey)", ident)
+        try query.filter(pivot, Local.foreignIdKey, id)
```

Comment:
Did you read the blurb about local and foreign I wrote in the PR comment? It _is_ super confusing. I think I might have an idea to make it simpler though.